enterprise/coderd/provisionerdaemons.go

```diff
@@ -221,6 +226,34 @@ func (api *API) provisionerDaemonServe(rw http.ResponseWriter, r *http.Request)
 		slog.F("tags", tags),
 	)
 
+	authCtx := ctx
+	if r.Header.Get(codersdk.ProvisionerDaemonPSK) != "" {
+		//nolint:gocritic // PSK auth means no actor in request,
+		// so use system restricted.
+		authCtx = dbauthz.AsSystemRestricted(ctx)
+	}
+
+	// Create the daemon in the database.
+	daemon, err := api.Database.UpsertProvisionerDaemon(authCtx, database.UpsertProvisionerDaemonParams{
+		Name:         name,
+		Provisioners: provisioners,
+		Tags:         tags,
+		CreatedAt:    dbtime.Now(),
+		LastSeenAt:   sql.NullTime{Time: dbtime.Now(), Valid: true},
+		Version:      "", // TODO: provisionerd needs to send version
```

Comment:
I assume that it may be optional for some time as it is possible to run newer version of Provisioner Daemon with legacy provisionerd.

Reply:
Yep it's just optional for now as far as I know, just adding it for future.